clang-tools-extra/clangd/index/dex/Dex.h

```diff
@@ -121,6 +121,8 @@ class Dex : public SymbolIndex {
   llvm::DenseMap<std::pair<SymbolID, uint8_t>, std::vector<SymbolID>> Relations;
   std::shared_ptr<void> KeepAlive; // poor man's move-only std::any
   // Set of files which were used during this index build.
+  // Files and IdxContents are only populated for dynamic and background
+  // indexes, not static indexes.
```

Comment:
i am not sure if it's worthwhile to encode some state of the users into the library here. they're likely to get out-of-sync quickly, and I don't think they provide any value to the reader. `Dex` is independent of the concepts such as dyanmic/static index.

Reply:
That's a fair point.

My motivation with this patch is to avoid the sort of confusion I had when looking at https://github.com/llvm/llvm-project/pull/118324, where I was wondering if maybe the fact that some codepaths do not propagate a value for `IdxContents` is an oversight.

I revised the comment to avoid talking about the specifics of higher layers here, let me know if that looks better.